Neos.Flow/Classes/Mvc/Routing/Dto/ResolveContext.php

```diff
@@ -62,6 +63,14 @@ public function __construct(UriInterface $baseUri, array $routeValues, bool $for
         $this->routeValues = $routeValues;
         $this->forceAbsoluteUri = $forceAbsoluteUri;
         $this->uriPathPrefix = $uriPathPrefix;
+
+        // Only add base uri path for absolute uri, in case of relative uri the uri has to be relative to the given base uri
+        if ($forceAbsoluteUri && $baseUri->getPath()) {
```

Comment:
`$baseUri->getPath()` is not a boolean (see https://neos.readthedocs.io/en/stable/References/CodingGuideLines/PHP.html#php-in-general => "Prefer strict comparisons whenever possible")